api/envoy/config/cluster/v3/cluster.proto

```diff
@@ -1164,6 +1164,10 @@ message UpstreamConnectionOptions {
 
   // If set then set SO_KEEPALIVE on the socket to enable TCP Keepalives.
   core.v3.TcpKeepalive tcp_keepalive = 1;
+
+  // If true, Envoy will make the interface name of the local address for the upstream connection
+  // available for debugging purposes. Defaults to false due to performance concerns.
```

Comment:
What does it mean to for the name to be made "available for debugging purposes"?  From the code, it looks like you're adding a header to the upstream request that contains the interface name, but the comment here doesn't explain that that's what it does, which makes it hard to understand.

How about something like this:

```
// If enabled, adds a "local_interface_name" header containing the interface name of the local address
// for the upstream connection to all upstream requests. Defaults to false due to performance concerns.
bool add_local_interface_name_header_on_upstream_requests = 2;
```

Reply:
"available for debugging purposes" == attached to the stream info object. So not exposed in the response. The code filter code that adds it to the response headers is all for testing purposes.

Reply:
If we want to make this generic, the best I can think of is:

`If enabled, associates the interface name of the local address with the upstream connection. This can be used by extensions during processing of requests. The association mechanism is implementation specific. Defaults to false due to performance concerns.`

Reply:
I took verbatim because I thought it was really good. Hope you don't mind @yanavlasov